### PR TITLE
Re-order overview of governance sections

### PR DESCRIPTION
**Pull request by Zsailer**
This is small PR that does some re-ordering of the governance overview page.

While working on #226, we found some places where re-ordering some subsections in the overview would make the page more accurately reflect the flow of decision making, e.g. we moved the subprojects summary _above_ the distinguished contributors. 

We also moved the old governance page to an archive folder.

### CODE REVIEW

**Review by afshin — APPROVED** [review]

File: overview.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Jupyter Governance Overview
 
 :::{note}
-Jupyter transitioned from a [Benevolent Dictator For Life (BDFL) + Steering Council](governance.md) governance model to this current governance model in December 2022.
+Jupyter transitioned from a [Benevolent Dictator For Life (BDFL) + Steering Council](archive/governance.md) governance model to this current governance model in December 2022.
```
Comment (jasongrout) [inline overview.md:4]: A change also needs to be done in the _toc.yml: https://github.com/jupyter/governance/pull/226/files#diff-3cccf3dee374305b8a8af99ba002fba0bf9932305cb9cbdcc81c962cbf54084f

**Comment by jasongrout** [comment]
Can we also add [this change](https://github.com/jupyter/governance/pull/226/files#diff-7a7987404e86a043b133f56a86fd282421f207557d254dc43ac4bb0bae8e7915R23) and [this change](https://github.com/jupyter/governance/pull/226/files#diff-7a7987404e86a043b133f56a86fd282421f207557d254dc43ac4bb0bae8e7915R31)?

**Comment by Zsailer** [comment]
@jasongrout done in https://github.com/jupyter/governance/pull/228/commits/9f60d4845f327af6bd03f0877c9fa51754425e65

File: overview.md (hunk 1 above)
Comment (Zsailer) [reply overview.md:4]: Done in https://github.com/jupyter/governance/pull/228/commits/9f60d4845f327af6bd03f0877c9fa51754425e65

**Review by jasongrout — APPROVED** [review]

**Pull request merged**